### DISCUSSION TITLE
Set default MTU for all providers, not just clab

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1544] [Features]
OSPF validation in integration/lag/01-l3-lag.yml fails for Cumulus_NVUE on Libvirt, because no default MTU is defined for that case on both FRR and Cumulus side. It works on clab

Wouldn't it be better if Netlab would define the default MTU for all providers, instead of only clab?

**Comment by DanPartelly** [comment]
I believe that yes, it would be better. 

> Wouldn't it be better if Netlab would define the default MTU for all providers, instead of only clab?

**Comment by ddutt** [comment]
Depends on what actual devices do. I thought they all set default mtu to 1500 as part of the nos itself.

**Reply by jbemmel** [reply]
No, at least some don't - in particular Cumulus defaults to 9216 (or some high value like that) which caused the OSPF test with FRR to fail

**Comment by ipspace** [comment]
The lad doth protest too much, methinks ;)

Let's look at the OSPF test results: https://tests.netlab.tools/_html/coverage.ospf.ospfv2

* The initial OSPF tests set the MTU as I had problems with some devices (probably Cumulus) -- https://github.com/ipspace/netlab/blob/dev/tests/integration/ospf/ospfv2/01-network.yml
* The "route import" OSPF tests do not set the link MTU (https://github.com/ipspace/netlab/blob/dev/tests/integration/ospf/ospfv2/10-import.yml), but they still pass on all supported devices. It is true, though, that I changed Cumulus from containerlab to libvirt because Cumulus 4.x container kept crashing my server.

It looks like most devices use the 1500 MTU as the default, so we have yet again a problem with a quirk of one of the platforms and you're trying to make it a "netlab should set a provider default" problem. There is a way to set the default MTU per device per provider (I happen to know it works because we're doing it for the **cumulus** device); please feel free to do that for Cumulus 5.x and document what you did in the caveats.

**Reply by jbemmel** [reply]
What I meant was: Adjust the MTU default setting **for Cumulus** to apply to all providers, not just clab

It needs the setting for all providers

**Reply by ipspace** [reply]
> What I meant was

Next time, please make the title and the description more explicit.

> Adjust the MTU default setting for Cumulus to apply to all providers, not just clab

That has already been done for "cumulus" device. Of course it's perfectly fine to do the same for the "cumulus_nvue" device.
